### ISSUE TITLE
reenable orcid as login method?

### ISSUE THREAD

**Issue by nicolasfranck** [DMPRoadmap/roadmap#2196]
**Expected behaviour:**

login with orcid provider should be possible

**Actual behaviour:**

While everything is present in the current code to make authentication with orcid possible,
the authentication is only used to link a logged in user to his/her orcid. No attempt is
made to link the hash `auth` to a user:

https://github.com/DMPRoadmap/roadmap/blob/development/app/controllers/users/omniauth_callbacks_controller.rb#L39

It appears that this worked in the past (see https://github.com/DMPRoadmap/roadmap/issues/628), but has been removed.

Even attempts have been removed: https://github.com/DMPRoadmap/roadmap/pull/1401

Is there a reason for this?

**Steps to reproduce:**

**Comment by NiklasZimmer** [comment]
From the side of UCT Libraries we would be very interested in further developments in this regard. At first glance at least, it seems to go against the whole notion of providing machine-actionable DMPs if the platform doesn't fully integrate with a widely used and free researcher-ID system such as ORCiD.

**Comment by briri** [comment]
Hi @nicolasfranck,

We opted to not incorporate login via ORCID at this point because they rarely pass important information back to Roadmap like email an email address or organizational affiliation (the user needs to go into their ORCID account and explicitly make those things 'public'). This makes it difficult for the system to do things like match an ORCID login with an existing user account. This is why we only allow the user to link to ORCID from the profile page (we already know their email, etc.)

We do not currently have plans to revisit this functionality in the near future. If you'd like to propose a workflow for using ORCID auth or send us a PR we would be happy to discuss further.

**Comment by NiklasZimmer** [comment]
Thanks @briri for this explanation, that makes sense to me!

**Comment by nicolasfranck** [comment]
@briri yes, but the same applies to shibboleth where the user can choose not to send certain attributes to the application server. For other applications we often use the member api so users do not have to make their primary email address available for the whole world. It is a matter of agreement.

But of course the application needs to deal with situations where certain required attributes are not returned (as is the case for shibboleth). Show the user a flash message that explains why authentication failed?
